script/jenkins-pr-job.sh

```diff
@@ -0,0 +1,355 @@
+# License: see ../LICENSE
+
+[[ -r /local/usr/ruby/.profile ]] && . /local/usr/ruby/.profile
+
+# some global vars to remember - don't wanna re-compute them in each function
+[[ -z ${WORKSPACE} ]] && WORKSPACE=${PWD}
+DATADIR=${WORKSPACE}/tmp/db # need to be absolute !
+typeset PR_IDF="${WORKSPACE}/tmp/PULLID_${ghprbPullId}"
+typeset BR_STATE_DIR=${JENKINS_HOME}/branch_states
+
+
+typeset PGBIN=${ pg_config --bindir ; }
+export PGDATA=${DATADIR}/pgsql/main PGHOST=${DATADIR}/var PGPORT=0
+#	dueto ontohub design flaw, ontohub wanna play DBA
+PG_DBA=postgres
+
+RDATA=${DATADIR}/redis RPID=0 RPORT=0
+#	for sidekiq
+export RSOCK="unix://${RDATA}/redis.sock" REDIS_PROVIDER=RSOCK
+
+EDATA=${DATADIR}/esearch EPORT=0
+
+HPID=0
+
+if [[ ${ uname -s ; } == 'SunOS' ]]; then
+	typeset SED=gsed DATE=gdate
+else
+	typeset SED=sed DATE=date
+fi
+
+typeset -T LogObj_t=(
+	# AnsiColor Plugin detects simple colors/"sequences", only.
+    typeset -Sh 'Color for info messages' GREEN='36';
+    typeset -Sh 'Color for warning messages' BLUE='34';
+    typeset -Sh 'Color for fatal messages' RED='31';
+    function log {
+        print -u2 "\E[1m\E[$2m$3\E[0m"
+    }
+    typeset -Sfh ' log a message to stderr' log
+    function info {
+        _.log "INFO" ${_.GREEN} "$*"
+    }
+    typeset -Sfh ' log a info message to stderr' info
+    function warn {
+        _.log "WARN" ${_.BLUE} "$*"
+    }
+    typeset -Sfh ' log a warning message to stderr' warn
+    function fatal {
+        _.log "FATAL" ${_.RED} "$*"
+    }
+    typeset -Sfh ' log a fatal error message to stderr' fatal
+    function printMarker {
+        typeset COLOR="$1"
+        print -f '\E[1;%sm----------------------------------------------------------------------------\E[0m\n' "${COLOR:-${_.GREEN}}"
+    }
+    typeset -Sfh ' print a marker line to stdout' printMarker
+)
+LogObj_t Log
+
+function tearDown {
+	${PGBIN}/pg_ctl -s -m fast stop
+	(( RPID )) && kill -9 ${RPID}
+	(( HPID )) && kill -9 ${HPID}
+	[[ -s ${EDATA}/pid ]] && kill -9 $(<${EDATA}/pid)
+	rm -f ${PR_IDF}
+	return 0
+}
+
+function prepareDatabase {
+	integer PORT=${EXECUTOR_NUMBER} C
+	(( PGPORT=PORT*10 + 15000 ))
+	(( C=PGPORT+10-1 ))
+	Log.warn "Used port range: ${PGPORT}..$C"
+
+	Log.info 'Setting up Postgres DB ...'
+	rm -rf ${RDATA} ${PGHOST} ${PGDATA}
+	mkdir -p ${RDATA} ${PGHOST} ${PGDATA} || exit 1
+
+    if [[ ! -x ${PGBIN}/initdb ]]; then
+		print -u2 "Uhmm '${PGBIN}/initdb' is not executable!"
+		return 1
+	fi
+    
+    ${PGBIN}/pg_ctl init -s -w -m fast \
+		-o "-E UTF8 --locale=en_US.UTF-8 --username=${PG_DBA}"
+	PROP='unix_socket_directories'
+	${SED} -i -e "/^#${PROP}/ s,^.*,${PROP} = '${DATADIR}/var'," \
+		-e "/^#port/ s,^.*,port = ${PGPORT}," \
+		${PGDATA}/postgresql.conf
+
+	${PGBIN}/pg_ctl -s -w start || return $?
+	print -n 'create user ontohub;
+create database ontohub OWNER=ontohub;
+create database ontohub_test OWNER=ontohub;
+create database ontohub_development OWNER=ontohub;
```

Comment:
We only need the database `ontohub_test` on the CI system because we only run the `test` environment.